ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -0,0 +1,236 @@
+        String body,
+        Map<String, String> parameters,
+        Map<Integer, ModelTensors> tensorOutputs
+    ) {
+        if (Strings.isBlank(body)) {
+            log.error("Remote model response body is empty!");
+            if (executionContext.getExceptionHolder().get() == null)
+                executionContext
+                    .getExceptionHolder()
+                    .compareAndSet(null, new OpenSearchStatusException("No response from model", RestStatus.BAD_REQUEST));
+        } else {
+            if (statusCode < HttpStatus.SC_OK || statusCode > HttpStatus.SC_MULTIPLE_CHOICES) {
+                log.error("Remote server returned error code: {}", statusCode);
+                if (executionContext.getExceptionHolder().get() == null)
+                    executionContext
+                        .getExceptionHolder()
+                        .compareAndSet(null, new OpenSearchStatusException(REMOTE_SERVICE_ERROR + body, RestStatus.fromCode(statusCode)));
+            } else {
+                try {
+                    ModelTensors tensors = processOutput(body, connector, scriptService, parameters, mlGuard);
+                    tensors.setStatusCode(statusCode);
+                    tensorOutputs.put(executionContext.getSequence(), tensors);
+                } catch (Exception e) {
+                    log.error("Failed to process response body: {}", body, e);
+                    if (executionContext.getExceptionHolder().get() == null)
+                        executionContext
+                            .getExceptionHolder()
+                            .compareAndSet(null, new MLException("Fail to execute predict in aws connector", e));
+                }
+            }
+        }
+    }
+
+    private void reOrderTensorResponses(Map<Integer, ModelTensors> tensorOutputs) {
+        List<ModelTensors> modelTensors = new ArrayList<>();
+        TreeMap<Integer, ModelTensors> sortedMap = new TreeMap<>(tensorOutputs);
+        log.debug("Reordered tensor outputs size is {}", sortedMap.size());
+        if (tensorOutputs.size() == 1) {
+            // batch API case
+            ModelTensors singleTensor = tensorOutputs.get(0);
+            int status = singleTensor.getStatusCode();
+            if (status == HttpStatus.SC_OK) {
+                modelTensors.add(singleTensor);
+                actionListener.onResponse(modelTensors);
+            } else {
+                actionListener.onFailure(buildOpenSearchStatusException(singleTensor));
+            }
+        } else {
+            // non batch API. This is to follow the previously code logic. Previously when making multiple requests to remote model,
+            // either one fails, we will return a failure response.
+            OpenSearchStatusException openSearchStatusException = null;
+            for (Map.Entry<Integer, ModelTensors> entry : sortedMap.entrySet()) {
+                if (entry.getValue().getStatusCode() < HttpStatus.SC_OK
```

Comment:
Do we need to check it again? If one response is error, we will set `exceptionHolder` and won't get into this function. Seems we can remove line 161-165